### PR TITLE
Refactor rename callbacks

### PR DESCRIPTION
**Pull request by benson31**
The existing naming scheme for callbacks is highly redundant. This fixes that. Depends on #1118. 

1. All callbacks are added to `namespace callback`.

2. To avoid a name collision with `namespace callback`, the base class is now `callback_base`.

3. Derived classes are changed as `s/lbann_callback_\(.*\)/\1/`.

4. Builder functions are renamed as `s/build_callback_\(.*\)_from_pbuf/build_\1_callback_from_pbuf/`. This doesn't really matter; it just sounds better to me.

5. Callback files are renamed as `s/callback_\(.*\)/\1/`.

### CODE REVIEW

**Comment by benson31** [comment]
Since #1118 merged, this has now been updated to reflect that. It's ready for review.

File: include/lbann/callbacks/io.hpp [hunk 1]
```diff
@@ -37,21 +37,22 @@
 #include <vector>
 
 namespace lbann {
+namespace callback {
 
 /**
  * Print information on the amount of IO that layers do.
  */
-class lbann_callback_io : public lbann_callback {
+class io : public callback_base {
```
[the remark this answers is not in the document]
Comment (benson31) [reply include/lbann/callbacks/io.hpp]: If I'm already renaming callbacks, I may as well rename it. Thoughts on what it should be? Best I got is `since_it_only_does_stuff_to_io_layers_it_should_not_be_a_full_fledged_callback_but_rather_part_of_the_input_layer`, but that seems long and... opinionated...

File: include/lbann/callbacks/mixup.hpp [hunk 2]
```diff
@@ -54,11 +55,11 @@ namespace lbann {
  *
  * The recommended default alpha (from the paper) is 0.4.
  */
-class callback_mixup : public lbann_callback {
+class callback_mixup : public callback_base {
```
[the remark this answers is not in the document]
Comment (benson31) [reply include/lbann/callbacks/mixup.hpp]: Good catch. Now the question is "why did `sed` not get this one?"...

File: include/lbann/callbacks/print.hpp [hunk 3]
```diff
@@ -32,18 +32,19 @@
...
 
 namespace lbann {
+namespace callback {
 
 /** Periodically print computational results.
  *  Prints average objective function value and metric scores after
  *  each training epoch and evaluation.
  */
-class lbann_callback_print : public lbann_callback {
+class print : public callback_base {
```
[the remark this answers is not in the document]
Comment (benson31) [reply include/lbann/callbacks/print.hpp]: `print_statistics`?

File: include/lbann/callbacks/io.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndryden) [reply include/lbann/callbacks/io.hpp]: `monitor_io` maybe?

File: include/lbann/callbacks/print.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (ndryden) [reply include/lbann/callbacks/print.hpp]: Sure.

File: include/lbann/callbacks/io.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (benson31) [reply include/lbann/callbacks/io.hpp]: Done.

File: include/lbann/callbacks/mixup.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (benson31) [reply include/lbann/callbacks/mixup.hpp]: Fixed.

File: include/lbann/callbacks/print.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (benson31) [reply include/lbann/callbacks/print.hpp]: Done.

**Comment by ndryden** [comment]
I think we may need to update some of the files using the Python frontend due to the renamed callbacks (mostly print).

**Comment by benson31** [comment]
I'm not sure since I didn't change their protobuf message names. If you think I should update those, too, I can, and then, yes, we would probably have to do that.

**Comment by ndryden** [comment]
Oh, nevermind then.

**Pull request merged**